ArduPlane/GCS_Mavlink.cpp

```diff
@@ -791,7 +791,7 @@ void GCS_MAVLINK_Plane::handleMessage(mavlink_message_t* msg)
         case MAV_CMD_DO_SET_HOME: {
             result = MAV_RESULT_FAILED; // assume failure
             if (is_equal(packet.param1, 1.0f)) {
-                plane.init_home();
+                plane.set_home_persistently(AP::gps().location());
```

Comment:
Unrelated to this PR: doesn't this mean that in this case this will always respond with a failure?

Reply:
Unrelated, but a bug.  I'll push a patch on top which resets `result`.